set.mm

```diff
@@ -412491,6 +412575,135 @@ about equation (2) in section I.1 in [Bollobas] p. 4.  (Contributed by
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
   Walks, paths and cycles
 #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+
+  A "walk" in a graph is usually defined for simple graphs, multigraphs or even
+  pseudographs as "alternating sequence of vertices and edges x0 , e1 , x1 ,
+  e2 , ... , e(l) , x(l) where e(i) = x(i-1)x(i), 0<i<=l.", see definition of
+  [Bollobas] p. 4, or "A walk (of length k) in a graph is a non-empty
+  alternating sequence v_0 e_0 v_1 e_1 ... e_k-1 v_k of vertices and edges in G
+  such that e_i = { v_i , v_i+1 } for all i < k.", see definition of [Diestel]
+  p. 10.
+
+  Formalizing these definitions (mainly by representing the indexed vertices
+  and edges by functions), a walk is represented by two mappings f from
+  { 1 , ... , n } and p from { 0 , ... , n }, where f enumerates the (indices
+  of the) edges (e is a third function enumerating the edges within the graph,
+  not within the walk), and p enumerates the vertices, see ~ df-wlks .  Hence a
+  walk (of length n) is represented by the following sequence:
+
+  p(0) e(f(1)) p(1) e(f(2)) ... p(n-1) e(f(n)) p(n).
+
+  Based on this definition of ` Walks `, the class of all walks, more
+  restrictive constructs are defined:
+
+  * ` Trails `: A "walk is called a _trail_ if all its edges are distinct.",
+    see Definition of [Bollobas] p. 5, i.e., f(i) =/= f(j) if i =/= j.
+
+  * ` Paths `: A _path_ is a walk whose vertices except the first and the last
+    vertex are distinct, i.e., p(i) =/= p(j) if i =/= j, except i = 0 and
```

Comment:
if i < j, except possibly when (i , j) = (0, n)
OR
if i =/= j, except possibly when (i , j) = (0, n) or (n, 0)
OR
(you may write couples as <. i , j >. instead of (i, j), I don't know what is preferred in comments)